src/modules/Game.class.js

```diff
@@ -5,64 +5,405 @@
+            moveTile++;
+          }
+
+          if (
+            !didJustMerged &&
+            tile - moveTile - 1 >= 0 &&
+            this.state[tile - moveTile - 1][column] === this.state[tile][column]
+          ) {
+            moveTile++;
+            merge = true;
+          }
+
+          if (moveTile > 0) {
+            if (!didJustMerged && merge) {
+              didJustMerged = true;
+
+              const newNumber = this.state[tile - moveTile][column] * 2;
+
+              this.state[tile - moveTile][column] = newNumber;
+
+              this.score += newNumber;
+            } else {
+              didJustMerged = false;
+
+              this.state[tile - moveTile][column] = this.state[tile][column];
+            }
+
+            this.state[tile][column] = 0;
+
+            didTilesMove = true;
+          }
+        }
+      }
+    }
+
+    if (didTilesMove) {
+      this.createRandomTile();
+    }
+
+    return didTilesMove;
+  }
+
+  moveDown() {
+    if (this.status !== Game.Status.playing) {
+      return;
+    }
+
+    this.saveState();
+
+    let didTilesMove = false;
+
+    for (let column = 0; column < 4; column++) {
+      let didJustMerged = false;
+
+      for (let tile = 2; tile >= 0; tile--) {
+        if (this.state[tile][column] > 0) {
+          let moveTile = 0;
+          let merge = false;
+
+          while (
+            tile + moveTile + 1 < this.state.length &&
+            this.state[tile + moveTile + 1][column] === 0
+          ) {
+            moveTile++;
+          }
+
+          if (
+            !didJustMerged &&
+            tile + moveTile + 1 < this.state.length &&
+            this.state[tile + moveTile + 1][column] === this.state[tile][column]
+          ) {
+            moveTile++;
+            merge = true;
+          }
+
+          if (moveTile > 0) {
+            if (!didJustMerged && merge) {
+              didJustMerged = true;
+
+              const newNumber = this.state[tile + moveTile][column] * 2;
+
+              this.state[tile + moveTile][column] = newNumber;
+
+              this.score += newNumber;
+            } else {
+              didJustMerged = false;
+
+              this.state[tile + moveTile][column] = this.state[tile][column];
+            }
+
+            this.state[tile][column] = 0;
+
+            didTilesMove = true;
+          }
+        }
+      }
+    }
+
+    if (didTilesMove) {
+      this.createRandomTile();
+    }
+
+    return didTilesMove;
+  }
+
+  saveState() {
+    return this.state.map((row) => row.slice());
+  }
+
+  createRandomTile() {
+    let row = this.randomNumber(3);
+
+    while (!this.state[row].includes(0)) {
+      row = this.randomNumber(3);
+    }
+
+    let cell = this.randomNumber(3);
+
+    while (this.state[row][cell] !== 0) {
+      cell = this.randomNumber(3);
+    }
+
+    const newState = this.state.map((currentRow) => currentRow.slice());
+
+    newState[row][cell] = this.generateCellValue();
+    this.state = newState;
```

Comment:
The `createRandomTile` method modifies the DOM directly. Ensure this method is called in an environment where the DOM is available, or consider abstracting DOM manipulation for better testability.